src/lib/builders/popover/create.ts

```diff
@@ -316,26 +304,14 @@ export function createPopover(args?: CreatePopoverProps) {
 	});
 
 	effect([open, activeTrigger, preventScroll], ([$open, $activeTrigger, $preventScroll]) => {
-		if (!isBrowser) return;
+		if (!isBrowser || !$open) return;
 
 		const unsubs: Array<() => void> = [];
 
-		if ($open) {
-			if (!$activeTrigger) {
-				tick().then(() => {
-					const triggerEl = document.getElementById(ids.trigger.get());
-					if (!isHTMLElement(triggerEl)) return;
-					activeTrigger.set(triggerEl);
-				});
-			}
-
-			if ($preventScroll) {
-				unsubs.push(removeScroll());
-			}
-
-			const triggerEl = $activeTrigger ?? document.getElementById(ids.trigger.get());
-			handleFocus({ prop: openFocus.get(), defaultEl: triggerEl });
+		if ($preventScroll) {
+			unsubs.push(removeScroll());
 		}
+		handleFocus({ prop: openFocus.get(), defaultEl: $activeTrigger });
```

Comment:
We "re-get" the trigger because it could change throughout use of the popover (via the ID prop mentioned in my previous comment)

Reply:
If the trigger changed throughout the use of the popover, the presupposition is that there was a trigger previously so `activeTrigger` is truthy. By that token we would never reach `document.getElementById`. Am I wrong?